### PR TITLE
Adjust documentation of `MakeDerivation`

### PR DESCRIPTION
**Pull request by TKuh**
~~And a small simplification of a check in `MakeDerivation`.~~

### CODE REVIEW

File: CAP/gap/Derivations.gi [hunk 1]
```diff
@@ -43,7 +43,7 @@ function( name, target_op, used_op_names_with_multiples_and_category_getters, we
     fi;
     # =#
     
-    if NumberArgumentsFunction( category_filter ) = 0 or NumberArgumentsFunction( category_filter ) > 1 then
```
Comment (zickgraf) [inline CAP/gap/Derivations.gi:46]: This is deliberate: Functions with a variable number of arguments have a negative `NumberArgumentsFunction`, e.g.
```
gap> NumberArgumentsFunction( {x...} -> 1 );
-1
```
and those should work here too, although I would not encourage this, which is why it's not mentioned in the error message.

Comment (TKuh) [reply CAP/gap/Derivations.gi:46]: Ah, thank you! Sorry for the inconvenience.

File: CAP/gap/Derivations.gd [hunk 2]
```diff
@@ -41,15 +41,18 @@ DeclareCategory( "IsDerivedMethod", IsAttributeStoringRep );
...
+#!  
 #!  This is given as a list of lists, where each sublist has as
 #!  first entry the name of an operation, as second entry an integer and as
-#!  third entry optionally a function. This function should accept the
-#!  category and return a category for which the operation in the first
-#!  entry must be installed for the derivation to be considered valid.
+#!  third entry either a function or <C>fail</C>. This function should accept the
+#!  category for which this derivation will be installed,
+#!  and return a category for which the operation in the first
+#!  entry is installed.
```
Comment (zickgraf) [inline CAP/gap/Derivations.gd:55]:
```suggestion
#!  entry must be installed for the derivation to be considered applicable.
```
It's okay if the operation in the first entry is not installed, then the derivation simply will not be applied.

File: CAP/gap/Derivations.gd [hunk 3]
```diff
@@ -58,7 +61,7 @@ DeclareCategory( "IsDerivedMethod", IsAttributeStoringRep );
 #!  The argument <A>func</A> contains the actual implementation of the
 #!  derived method.
 #!  The argument <A>category_filter</A> is a filter (or function) describing
-#!  which categories the derivation is valid for.  If it is valid
+#!  for which categories the derivation is valid for. If it is valid
```
Comment (zickgraf) [inline CAP/gap/Derivations.gd:64]: This sentence now contains "for" two times.

**Review by zickgraf — APPROVED** [review]
Thanks!

**Pull request merged**